### ISSUE TITLE
"Web3.py only accepts checksum addresses...Or, if you must accept lower safety, use Web3.toChecksumAddress(lower_case_address)."

### ISSUE THREAD

**Issue by arturgontijo** [singnet/snet-cli#73]
To reproduce:
```
$ snet set current_agent_at 0x1e89d9ed5bcc22f934af631cda771019081e57b2
set current_agent_at 0x1e89d9ed5bcc22f934af631cda771019081e57b2
```
Then running `snet client call [method] [json]`, I got this:
```
...
File "/usr/local/lib/python3.6/dist-packages/web3/utils/validation.py", line 123, in validate_address
    value,
web3.exceptions.InvalidAddress: ('Web3.py only accepts checksum addresses. The software that gave you this non-checksum address should be considered unsafe, please file it as a bug on their platform. Try using an ENS name instead. Or, if you must accept lower safety, use Web3.toChecksumAddress(lower_case_address).', '0x1e89d9ed5bcc22f934af631cda771019081e57b2')
```
But getting the address from etherscan (`0x1E89D9ed5bCC22F934AF631CdA771019081E57B2`), it works.

Question is...should we use this `lower safety` function or just warn user to change address format?

**Comment by astroseger** [comment]
I would propose to do following:
for contract address we should accept lower case address because ganache, at the moment, print contract address in lower case. 
for account address we should accept only checksum address.

**Comment by arturgontijo** [comment]
I think the best way will be always call `type_converter("address")(INPUT_ADDRESS)` before make a transaction that uses any `INPUT_ADDRESS`.
Maybe you're already doing this with MPE commands.

**Comment by stellarspot** [comment]
I run the command
```
snet service metadata_init service/service_spec DNNModelService 0x3b2b3c2e2e7c93db335e69d827f3cc4bc2a2a2cb --multipartyescrow 0x5c7a4290f6f8ff64c69eeffdfafc8644a4ec3a4e
```

where the 0x3b2b3c2e2e7c93db335e69d827f3cc4bc2a2a2cb is the second address from the Ganache output and it fails with error:
```
Error: payment_address parameter is not a valid Ethereum checksum address
```
If I use the address from the [test](https://github.com/astroseger/my-platform-tests/blob/master/test1/run_service.sh) 0x3b2b3C2e2E7C93db335E69D827F3CC4bC2A2A2cB  it passes.

Is it intentional behavior?

**Comment by astroseger** [comment]
yes... it is expected behavior. User should use only chacksum addresses. At least Ethereum actively promote it.

**Comment by astroseger** [comment]
@arturgontijo 
I would propose the following behavior.
1. For accounts we accept only in checksum format (it is more secure and we follow this line...)
2. For contracts we accept any format (we have it only in "snet set  current_registry_at/current_multipartyescrow_at/current_singularitynettoken_at")

**Comment by astroseger** [comment]
closed with #142
